nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -1,21 +1,146 @@
+
+    def __init__(self, file_path: FilePathType):
+        assert HAVE_OPENCV, INSTALL_MESSAGE
+        self.vc = cv2.VideoCapture(filename=file_path)
+        self.file_path = file_path
+        self._current_frame = 0
+        self._frame_count = None
+        self._movie_open_msg = "The Movie file is not open!"
+
+    def get_movie_timestamps(self):
+        """Return numpy array of the timestamps(s) for a movie file."""
+        ts2 = []
+        for no in tqdm(range(self.get_movie_frame_count()), desc="retrieving timestamps"):
+            success, frame = self.vc.read()
+            ts2.append(self.vc.get(cv2.CAP_PROP_POS_MSEC) / 1000)
+            if not success:
+                break
+        return np.array(ts2)
+
+    def get_movie_fps(self):
+        """Return the internal frames per second (fps) for a movie file"""
+        assert self.isOpened(), self._movie_open_msg
+        prop = self.get_cv_attribute("CAP_PROP_FPS")
+        return self.vc.get(prop)
+
+    def get_frame_shape(self) -> Tuple:
+        """Return the shape of frames from a movie file."""
+        frame = self.get_movie_frame(0)
+        if frame is not None:
+            return frame.shape
+
+    @property
+    def frame_count(self):
+        if self._frame_count is None:
+            self._frame_count = self._movie_frame_count()
+        return self._frame_count
+
+    @frame_count.setter
+    def frame_count(self, val):
+        self._frame_count = val
+
+    def get_movie_frame_count(self):
+        return self.frame_count
+
+    @staticmethod
+    def get_cv_attribute(attribute_name):
+        if int(cv2.__version__.split(".")[0]) < 3:  # pragma: no cover
+            return getattr(cv2.cv, "CV_" + attribute_name)
+        return getattr(cv2, attribute_name)
+
+    def _movie_frame_count(self):
+        """Return the total number of frames for a movie file."""
+        assert self.isOpened(), self._movie_open_msg
+        prop = self.get_cv_attribute("CAP_PROP_FRAME_COUNT")
+        return int(self.vc.get(prop))
+
+    @property
+    def current_frame(self):
+        return self._current_frame
+
+    @current_frame.setter
+    def current_frame(self, frame_number):
+        assert self.isOpened(), self._movie_open_msg
+        set_arg = self.get_cv_attribute("CAP_PROP_POS_FRAMES")
+        set_value = self.vc.set(set_arg, frame_number)
+        if set_value:
+            self._current_frame = frame_number
+        else:
+            raise ValueError(f"could not set frame no {frame_number}")
+
+    def get_movie_frame(self, frame_number: int):
+        """Return the specific frame from a movie."""
+        assert self.isOpened(), self._movie_open_msg
+        assert frame_number < self.get_movie_frame_count(), "frame number is greater than length of movie"
+        self.current_frame = frame_number
+        success, frame = self.vc.read()
+        self.current_frame = 0
+        return frame
```

Comment:
```suggestion
    def get_movie_frame(self, frame_number: int):
        """Return the specific frame from a movie."""
        assert self.isOpened(), self._movie_open_msg
        assert frame_number < self.get_movie_frame_count(), "frame number is greater than length of movie"
        initial_frame_number = self.current_frame
        self.current_frame = frame_number
        success, frame = self.vc.read()
        self.current_frame = initial_frame_number 
        return frame
```
Let's say you partially iterate (or explicitly set the current frame to some value), then explicitly ask for some other frame (before or after the point you stopped iterating); if you reset the current frame to zero, you erase the point at which the iteration stopped (or whatever value you were otherwise saved at). This instead restores it to act more like an in-place operation.

Reply:
yes, that's right, its safer to do this internally than expecting to set `self.current_frame=no` to the previous value.